app/src/main/java/com/andrew/tasky/agenda/presentation/adapters/PhotoItemAdapter.kt

```diff
@@ -22,11 +20,19 @@ class PhotoItemAdapter(
 
     companion object : DiffUtil.ItemCallback<EventPhoto>() {
         override fun areItemsTheSame(oldItem: EventPhoto, newItem: EventPhoto): Boolean {
-            return oldItem == newItem
+            return if (oldItem is EventPhoto.Local && newItem is EventPhoto.Local) {
+                oldItem.id == newItem.id
+            } else if (oldItem is EventPhoto.Remote && newItem is EventPhoto.Remote) {
+                oldItem.key == newItem.key
+            } else false
```

Comment:
Here you would also share the identifier and just have one key attribute for both (in the constructor of the `EventPhoto` sealed class)